google/cloud/security/inventory/inventory_loader.py

```diff
@@ -191,98 +191,56 @@ def _build_pipelines(cycle_timestamp, configs, **kwargs):
-            kwargs.get(project_dao_name)
-        ),
-    ]
-
-    if configs.get('inventory_groups'):
-        if util.can_inventory_groups(configs):
-            admin_api_client = ad.AdminDirectoryClient()
-            pipelines.extend([
-                load_groups_pipeline.LoadGroupsPipeline(
-                    cycle_timestamp,
-                    configs,
-                    admin_api_client,
-                    dao),
-                load_group_members_pipeline.LoadGroupMembersPipeline(
+
+    inventory_pipeline_configs = file_loader.read_and_parse_file(
+        'google/cloud/security/inventory/inventory_pipeline_configs.yaml')
+    
+    # first pass: map all the pipelines to their own nodes,
+    # regardless if they should run or not
+    map_of_all_pipeline_nodes = {}
+    for i in inventory_pipeline_configs:
+        map_of_all_pipeline_nodes[i.get('resource')] = PipelineNode(
+            i.get('resource'), i.get('should_run'), i.get('module_name'))
+
+    # another pass: build the dependency tree by setting the parents
+    # correctly on all the nodes
+    for i in inventory_pipeline_configs:
+        parent_name = i.get('depends_on')
+        if parent_name is not None:
+            parent_node = map_of_all_pipeline_nodes[parent_name]
+            map_of_all_pipeline_nodes[i.get('resource')].parent = parent_node
+
+    root = map_of_all_pipeline_nodes.get('organizations')
+
+    # If child pipeline is true, then all parents will become true.
+    # Even if the parent(s) is(are) false.
+    # Manually traverse the parents since anytree walker api doesn't make sense.
+    for node in anytree.iterators.PostOrderIter(root):
+        if node.should_run:
+            while node.parent is not None:
+                node.parent.should_run = node.should_run
+                node = node.parent
+
+    print anytree.RenderTree(root, style=anytree.AsciiStyle()).by_attr('resource_name')
+    print anytree.RenderTree(root, style=anytree.AsciiStyle()).by_attr('should_run')
+
+    # Now, we have the true state of whether a pipeline should be run or not.
+    # Get a list of pipeline instances that will actually be run.
+    pipelines_to_run = []
+    for node in anytree.iterators.PreOrderIter(root):
+        if node.should_run:
+            module_name = 'google.cloud.security.inventory.pipelines.{}'.format(node.module_name)
+            module = importlib.import_module(module_name)
+            class_name = node.module_name.title().replace('_', '')
+            pipeline_class = getattr(module, class_name)
+            pipelines_to_run.append(
+                pipeline_class(
                     cycle_timestamp,
                     configs,
-                    admin_api_client,
-                    dao)
-            ])
-        else:
-            raise inventory_errors.LoadDataPipelineError(
-                'Unable to inventory groups with specified arguments:\n%s',
-                configs)
+                    crm_v1_api_client,
```

Comment:
I think I might've missed something but it's not clear to me how each pipeline knows which GCP API client or dao class to use? (e.g. organization pipeline uses organization_dao, project pipeline uses project_dao, etc.)